examples/awaits.rb

```diff
@@ -38,47 +46,46 @@
       false
     end
   end
-
-  # Let the user know we're  ready and listening..
-  event.respond 'Guess a number between 1 and 10..'
+  event.respond "My number was: `#{magic}`."
 end
 
-# Above we used the provided User#await method to easily set up
+# Above we used the provided User#await! method to easily set up
 # an await for a follow-up message from a user.
 # We can also manually register an await for specific kinds of events.
 # Here, we'll write a command that shows the current time and allows
 # the user to delete the message with a reaction.
-# We'll be using Bot#add_await to do this:
-# https://www.rubydoc.info/gems/discordrb/Discordrb%2FBot:add_await
+# We'll be using Bot#add_await! to do this:
+# https://rubydoc.info/github/swarley/discordrb/Discordrb%2FBot:add_await!
 
 # the unicode ":x:" emoji
 CROSS_MARK = "\u274c"
 
 bot.message(content: '!time') do |event|
-  # Send a message, and store a reference to it that we can issue a delete request later
+  # Send a message, and store a reference to it that we can add the reaction.
   message = event.respond "The current time is: #{Time.now.strftime('%F %T %Z')}"
 
   # React to the message to give a user an easy "button" to press
   message.react CROSS_MARK
 
   # Add an await for a ReactionAddEvent, that will only trigger for reactions
-  # that match our CROSS_MARK emoji. This time, I'm using interpolation to make the
-  # await key unique for this event so that multiple awaits can exist.
-  bot.add_await(:"delete_#{message.id}", Discordrb::Events::ReactionAddEvent, emoji: CROSS_MARK) do |reaction_event|
-    # Since this code will run on every CROSS_MARK reaction, it might not
-    # be on our time message we sent earlier. We use `next` to skip the rest
-    # of the block unless it was our message that was reacted to.
-    next true unless reaction_event.message.id == message.id
-
-    # Delete the matching message.
-    message.delete
+  # that match our CROSS_MARK emoji. To prevent the bot from cluttering up threads, we destroy the await after 30 seconds.
+  bot.add_await!(Discordrb::Events::ReactionAddEvent, timeout: 30) do |reaction_event|
+    # Ignore all ReactionAddEvents not affecting our message or reactions not being the CROSS_MARK
+    next if message.id != reaction_event.message.id || reaction_event.emoji.to_reaction != CROSS_MARK
+
+    # A CROSS_MARK reaction was added.
+    message.delete # Delete the bot message
+    event.message.delete # Delete the user message
   end
```

Comment:
I just merged an older PR I made that cleans this up considerably :)
```suggestion
  bot.add_await!(Discordrb::Events::ReactionAddEvent, message: message, emoji: CROSS_MARK, timeout: 30) do |reaction_event|
    message.delete # Delete the bot message
  end
```

Reply:
Didn't know that was possible, but that looks waaay better ^^